packages/radix-ui-themes/src/props/prop-def.ts

```diff
@@ -68,10 +68,11 @@ type PropDef<T = any> = RegularPropDef<T> | ResponsivePropDef<T>;
 
 // prettier-ignore
 type GetPropDefType<Def> =
-	  Def extends BooleanPropDef ? (Def extends ResponsivePropDef ? Responsive<boolean> : boolean)
-  : Def extends StringPropDef ? (Def extends ResponsivePropDef ? Responsive<string> : string)
+    Def extends StringPropDef ? (Def extends ResponsivePropDef ? Responsive<string> : string)
+  : Def extends BooleanPropDef ? (Def extends ResponsivePropDef ? Responsive<boolean> : boolean)
   : Def extends ReactNodePropDef ? (Def extends ResponsivePropDef ? Responsive<React.ReactNode> : React.ReactNode)
-  : Def extends EnumOrStringPropDef<infer Type> ? (Def extends ResponsivePropDef<infer Type extends string> ? Responsive<Union<string, Type>> : Type)
+  : Def extends EnumOrStringPropDef<infer Type> ?
+    Def extends ResponsivePropDef<infer Type extends string> ? Responsive<Union<string, Type>> : Type
```

Comment:
We have something weird with this inference here. Was getting a responsive boolean type on `maxWidth` prop when it’s default was reassigned: 

![Screenshot 2024-02-27 at 11 27 14](https://github.com/radix-ui/themes/assets/8441036/1edd3774-9eef-4851-8c2e-b33f318c8f41)

I couldn't find the root cause, but reordering the order in which the prop def type is inferred fixed that.

Reply:
I think I figured out what the issue is with this, let me know if you want to hop on a call.